File: frontend/src/services/AuthService.js
// src/services/AuthService.js

import axios from "axios";

const API_URL = "http://localhost:8087/api/auth"; // ✅ Make sure this matches your Spring Boot server port

export const signup = (signupRequest) => {
  return axios.post(`${API_URL}/signup`, signupRequest);
};

export const login = (loginRequest) => {
  return axios.post(`${API_URL}/login`, loginRequest);
};

File: backend/src/main/java/com/BillingApp/services/AuthService.java
package com.BillingApp.services;

import com.BillingApp.DTO.AuthRequest;
import com.BillingApp.DTO.AuthResponse;
import com.BillingApp.model.SignupRequest;

public interface AuthService {
    AuthResponse login(AuthRequest request);
    void signup(SignupRequest request);
}
